electron/main/ipcEvents.ts

```diff
@@ -69,28 +70,60 @@ export default class IpcEvents {
 
     private vanish() {
         ipcMain.on(IpcEvent.Vanish, () => {
-            log.info('Vanishing overlay');
-
-            const windowBounds = this.overlay?.getBounds() as Rectangle;
-
-            log.info('Saved window state', windowBounds);
-
-            const theme = this.store.get('savedWindowState').theme;
-
-            const data: WindowState = {
-                x: windowBounds.x,
-                y: windowBounds.y,
-                width: windowBounds.width,
-                height: windowBounds.height,
-                isClickThrough: true,
-                isTransparent: true,
-                theme,
-            };
-
-            this.store.set('savedWindowState', data);
+            if (!this.store.get('settings').isOpen && this.store.get('savedWindowState').isTransparent) {
+                // DISABLING VANISH CASE
+                // Settings are CLOSED and the Window IS transparent
+                log.info('Disabling Vanish');
+                this.store.set<typeof StoreKeys.SavedWindowState>('savedWindowState', {
+                    ...this.store.get('savedWindowState'),
+                    isClickThrough: false,
+                    isTransparent: false,
+                });
+
+                this.overlay?.setIgnoreMouseEvents(false);
+                this.overlay?.webContents.send(IpcEvent.ShowApp);
+            } else if (!this.store.get('settings').isOpen && !this.store.get('savedWindowState').isTransparent) {
+                // ENABLING VANISH CASE
+                // Settings are CLOSED and Window IS NOT transparent
+                log.info('Vanishing overlay');
+
+                const windowBounds = this.overlay?.getBounds() as Rectangle;
+
+                log.info('Saved window state', windowBounds);
+
+                const theme = this.store.get('savedWindowState').theme;
+
+                const data: WindowState = {
+                    x: windowBounds.x,
+                    y: windowBounds.y,
+                    width: windowBounds.width,
+                    height: windowBounds.height,
+                    isClickThrough: true,
+                    isTransparent: true,
+                    theme,
+                };
+
+                this.store.set('savedWindowState', data);
+
+                this.overlay?.setIgnoreMouseEvents(true);
+                this.overlay?.webContents.send(IpcEvent.Vanish);
+            } else {
+                // Settings are OPEN and Vanish Call is activated
+                // Just logged, nothing happens for the User
+                log.info('OOPS! Out of Case, how did that happen?');
```

Comment:
please update this message to something like "settings open and vanish called"

just so that it is meaningful when the logs are checked